src/main/java/no/unit/nva/model/Publication.java

```diff
@@ -105,6 +105,25 @@ public void setDoiRequest(DoiRequest doiRequest) {
         this.doiRequest = doiRequest;
     }
 
+    /**
+     * Update a publication with the requested status change.
+     * @param requestedStatusChange Requested status change.
+     * @throws IllegalArgumentException Invalid status to change to.
+     * @throws IllegalStateException No DoiRequest exists.
+     * @see DoiRequestStatus
+     */
+    public void updateDoiRequestStatus(DoiRequestStatus requestedStatusChange) {
+        if (Objects.isNull(doiRequest)) {
```

Comment:
Static import would be prettier, I'd have modelled a NullObject, but that's me. I'd also probably have ended up with a mess of classes and some weird interface, so maybe just as well I didn't.

Reply:
I'll fix it in develop after merge. Thanks for the observation.